packages/lit-dev-content/site/blog/2022-07-19-lit-for-polymer-users.md

```diff
@@ -0,0 +1,935 @@
+
+```js
+render() {
+  let count = this.getMatchingItems(this.pattern);
+
+  return html`You have ${count} matches.`
+}
+```
+
+Like Polymer, Lit supports setting properties, attributes, and event handlers using expressions. Lit uses slightly different syntax, with prefixes instead of suffixes.
+
+Polymer:
+
+```html
+<-- property binding--no suffix -->
+<target-element property-binding=[[value]]><target-element>
+
+<!-- attribute binding adds a $ suffix -->
+<target-element attribute-binding$=[[value]]></target-element>
+
+<!-- boolean attribute binding adds a ? suffix -->
+<target-element hidden?=[[value]]></target-element>
+
+<!-- event handler uses on-[event-name] syntax -->
+<target-element on-click=[[handler]]></target-element>
+```
+
+Lit:
+
+```html
+<!-- property binding adds a "." prefix -->
+<target-element .property-binding=${value}><target-element>
+
+<!-- attribute binding--no prefix -->
+<target-element attribute-binding$=${value}></target-element>
+
+<!-- boolean attribute binding adds a ? prefix -->
+<target-element ?hidden=${value}></target-element>
+
+<!-- event handler uses @event-name syntax -->
+<target-element @click=${handler}></target-element>
+```
+
+In Lit, the handler can be either an unbound method, like `${this.clickHandler}` or an arrow function. Using an arrow function, you can close over other data or call a function with a different signature. For example:
+
+```html
+<input @change=${(e) => this.setValue(e.target.value)}>
+```
+
+For more information, see [Expressions](https://lit.dev/docs/templates/expressions/).
+
+
+### Two-way bindings
+
+In general, we recommend removing two-way bindings before migrating Polymer projects to Lit. For more information, see [Removing two-way binding](#removing-two-way-bindings).
+
+If you've replaced your two-way bindings with one-way bindings and event listeners, you should be able to migrate them fairly directly to Lit.
+
+If you're writing a Lit component to replace a parent component that used to use two-way binding to communicate with a Polymer child component, add a property expression to set the property, and an event listener to handle the `property-changed` event.
+
+
+Polymer:
+
+```js
+static get template() {
+  return html`
+    <polymer-child
+      childprop="[[parentprop]]"
+      on-childprop-changed="childpropChanged">
+    </polymer-child>
+  `;
+}
+
+childpropChanged(e) {
+  this.parentprop = e.detail.value;
+}
+```
+
+Lit:
+
+
+```js
+static template = html`
+  <polymer-child
+    .childprop=${this.parentprop}
+    @childprop-changed=${(e) => this.parentprop = e.detail.value}>
+  </polymer-child>
+`;
+```
+
+
+#### Conditional templates
+
+Polymer supports conditionals using the `dom-if` helper element.
+
+```html
+<template is="dom-if" if="{{condition}}">
+  <div>condition is true</div>
+</template>
+```
+
+In Lit, you can use a JavaScript conditional expressions. The conditional operator (or ternary) works well:
+
+```js
+html`${this.condition
```

Comment:
I thought we'd decided we should use `map` by default from now on, because it has some advantages over array.map (namely, being able to use other, non-array iterables).

Whereas `when` doesn't seem to add anything other than another import statement.